### PR TITLE
all: global find and replace of wasm-tools-go

### PR DESCRIPTION
**Pull request by Mossaka**
Contributes to #232.

### CODE REVIEW

File: wit/docs.go [hunk 1]
```diff
@@ -32,6 +32,6 @@
 // [WebAssembly Interface Type]: https://component-model.bytecodealliance.org/design/wit.html
 // [WebAssembly Component Model]: https://component-model.bytecodealliance.org/introduction.html
 // [wit-parser]: https://docs.rs/wit-parser/latest/wit_parser/
-// [source]: https://github.com/bytecodealliance/wasm-tools/tree/main/crates/wit-parser
+// [source]: https://go.bytecodealliance.org/wasm-tools/tree/main/crates/wit-parser
```
Comment (ydnar) [inline wit/docs.go:35]: Typo?

File: wit/wit.go [hunk 2]
```diff
@@ -433,7 +433,7 @@ func escape(name string) string {
 
 // A map of all [WIT keywords].
 //
-// [WIT keywords]: https://github.com/bytecodealliance/wasm-tools/blob/main/crates/wit-parser/src/ast/lex.rs#L524-L591
+// [WIT keywords]: https://go.bytecodealliance.org/wasm-tools/blob/main/crates/wit-parser/src/ast/lex.rs#L524-L591
```
Comment (ydnar) [inline wit/wit.go:436]: Typo?

File: wit/testdata_test.go [hunk 3]
```diff
@@ -495,7 +495,7 @@ func TestHasBorrowOnNamedTypes(t *testing.T) {
 			for _, td := range res.TypeDefs {
 				if td.Name == nil {
 					// Skip anonymous types here. This is tested on `TestHasBorrow`.
-					// https://github.com/bytecodealliance/wasm-tools-go/issues/167
+					// https://go.bytecodealliance.org/issues/167
```
Comment (lxfontes) [inline wit/testdata_test.go:498]: think we want to keep this one pointing directly to github

File: wit/resolve.go [hunk 4]
```diff
@@ -1076,7 +1076,7 @@ func (l *List) hasResource() bool         { return HasResource(l.Type) }
 // Future represents a WIT [future type], expected to be part of [WASI Preview 3].
 // It implements the [Node], [ABI], and [TypeDefKind] interfaces.
 //
-// [future type]: https://github.com/bytecodealliance/wit-bindgen/issues/270
+// [future type]: https://go.bytecodealliance.org/wit-bindgen/issues/270
```
Comment (lxfontes) [inline wit/resolve.go:1079]: think we want to keep this one pointing directly to github

**Comment by ydnar** [comment]
I think it might be worth doing a first pass of replacing `https://github.com/bytecodealliance/wasm-tools-go` with `https://github.com/bytecodealliance/go-modules` before finding and replacing `github.com/bytecodealliance/wasm-tools-go` with `go.bytecodealliance.org`.

**Comment by Mossaka** [comment]
Do we want to replace all the `https://github.com/bytecodealliance/wasm-tools-go` in CHANGELOG.md too?

**Comment by ydnar** [comment]
> Do we want to replace all the `https://github.com/bytecodealliance/wasm-tools-go` in CHANGELOG.md too?

Yes, this will pick up tags, releases, issues, PRs.

File: wit/docs.go [hunk 5]
```diff
@@ -32,6 +32,6 @@
 // [WebAssembly Interface Type]: https://component-model.bytecodealliance.org/design/wit.html
 // [WebAssembly Component Model]: https://component-model.bytecodealliance.org/introduction.html
 // [wit-parser]: https://docs.rs/wit-parser/latest/wit_parser/
-// [source]: https://github.com/bytecodealliance/wasm-tools/tree/main/crates/wit-parser
+// [source]: https://github.com/wasm-tools/tree/main/crates/wit-parser
```
Comment (ydnar) [inline wit/docs.go:35]: This looks broken.

File: wit/wit.go [hunk 6]
```diff
@@ -433,7 +433,7 @@ func escape(name string) string {
 
 // A map of all [WIT keywords].
 //
-// [WIT keywords]: https://github.com/bytecodealliance/wasm-tools/blob/main/crates/wit-parser/src/ast/lex.rs#L524-L591
+// [WIT keywords]: https://github.com/wasm-tools/blob/main/crates/wit-parser/src/ast/lex.rs#L524-L591
```
Comment (ydnar) [inline wit/wit.go:436]: Also broken.

**Review by ydnar — APPROVED** [review]

**Review by lxfontes — APPROVED** [review]

**Pull request merged**